### ISSUE TITLE
Auto-detect potential weather models based on DEM extents

### ISSUE THREAD

**Issue by jforthofer** [firelab/windninja#291]
Thinking about firelab/windninja-mobile#147, which may get done soon in a contract with GCS, it seems like it would make sense to implement code in windninja to test if a given DEM is within a weather model's areal coverage.  This could be useful in WindNinja desktop for other things, as well as the mobile server side.
@ksshannon or @nwagenbrenner, does this already exist?
In WindNinja desktop, we could potentially modify the weather model choices in the drop down box to only those that are valid for the user's DEM.  Hopefully this could be done fast enough so the GUI isn't bogged down, reading weather model extents from disk might be the issue here, or not?

**Comment by nwagenbrenner** [comment]
This doesn't exist yet as far as I know, but seems like a good idea. We could store a bounding box for each wx model format and check if the DEM falls within each one and modify the choices in the GUI based on that. We wouldn't need to read anything from disk, just store the bounding boxes for each model.

**Comment by jforthofer** [comment]
That's what I was thinking at first, but then I was thinking that if the DEM and weather models aren't in the same projection, they probably don't look like simple rectangles with straight sides.  So to do it right I think you'd have to do a polygon-in-polygon type test.  Does this sound correct?

**Comment by nwagenbrenner** [comment]
I think we could store the projection and geotransform for each model and warp the DEM to that projection to check if it falls within. Or just transform the corner points. @ksshannon would know the best way.

**Comment by ksshannon** [comment]
I'm not sure exactly how they are stored, but they are irregular.  Probably just dump a file in data/ and lazily load it.

http://www.emc.ncep.noaa.gov/mmb/namgrids/g212.12kmexp.jpg

**Comment by ksshannon** [comment]
LGTM, we just have to find all of them.  I am guessing most have similar extent, but different dx/dy?

**Comment by nwagenbrenner** [comment]
Yeah, I can get them all. Some are AK only, otherwise similar extents. Except GFS, which is the whole world so doesn't need to be tested anyway.

**Comment by jforthofer** [comment]
I guess I'm still not following.  If the DEM and weather model are in two different projections/coordinate systems, how do you compare them?  You can't just use corner points (even if you converted all the corners to lat/lon), because the edges aren't lines, they are curved.

**Comment by jforthofer** [comment]
Yeah.  If we didn't care about being too exact, we might be able to build a fast running lat/lon corner point based method that was very conservative (so it would lean toward saying the DEM is not in the bounds if you are near boundaries, like having a certain buffer built in or something).  You might lose some simulations near the boundaries, but that might be OK since it is probably a low occurrence.

**Comment by ksshannon** [comment]
I don't think we need to worry about implementation details, I think I can figure out how to do an exact comparison.  A footprint warped into a common should work, then ST_Contains(g1,g2).

**Comment by ksshannon** [comment]
Any objections to pushing to 3.4.x or 3.5.x?

**Comment by nwagenbrenner** [comment]
Fine with me too.
